src/protocol/sort/apply_sort/shuffle.rs

```diff
@@ -175,5 +213,42 @@ mod tests {
             // Shuffled output should have same inputs
             assert_eq!(hashed_output_secret, hashed_input);
         }
+
+        #[tokio::test]
+        async fn shuffle_vec_of_replicated() {
+            let original_match_keys = [
+                123_456_789,
+                234_567_890,
+                345_678_901,
+                456_789_012,
+                567_890_123,
+            ];
+            let input = original_match_keys.map(|x| get_bits::<Fp32BitPrime>(x, 32));
+            let world = TestWorld::new(QueryId);
+            let result = world
+                .semi_honest(input, |ctx, vec_of_vec_of_shares| async move {
+                    let perms = get_two_of_three_random_permutations(5, ctx.prss_rng());
+                    shuffle_shares(
+                        vec_of_vec_of_shares,
+                        (perms.0.as_slice(), perms.1.as_slice()),
+                        ctx,
+                    )
+                    .await
+                    .unwrap()
+                })
+                .await
+                .reconstruct();
+
+            let all_match_keys: HashSet<u32> = HashSet::from_iter(original_match_keys);
+
+            let mut match_keys_found = Vec::with_capacity(5);
+            for row in result {
+                let mk = u32::try_from(bits_to_value(&row)).unwrap();
+                assert!(all_match_keys.contains(&mk));
+                match_keys_found.push(mk);
+            }
```

Comment:
You should just be able to sort the output and check for equality with the input (as you do below).  Checking that each is found isn't necessary.